### PR TITLE
Fix submission modal displaying wrong labels for choices

### LINKED ISSUE #2969: Page goes blank when users click the eye like preview button from DATA>Table!
## Description

Page goes blank when users click the `eye like preview button` from `DATA>Table`.

## Steps to Reproduce

1. Open the project.
2. Go to `DATA>Table`.
3. Click the `eye like preview button`. The page goes blank.

## Expected behavior 

User should be able to preview the data that is present in their account.

## Actual behavior 

Page goes blank.

## Additional details

Viewing the `custom report`, `editing the submission` has no effect i.e. they work smoothly as expected. **Link to the community forum:** https://community.kobotoolbox.org/t/kobotoolbox-returning-a-blank-page-after-clicking-the-submitted-form/15288/9

### LINKED ISSUE #2971: Submission Modal displays wrong labels for choices with same names
## Description

When you have multiple choices questions and use the same `name` value for their choices, then the Submission Modal is displaying always the label from the first one.

## Additional details

Community link: https://community.kobotoolbox.org/t/data-table-open-view-no-longer-aligned-after-redeployment/15316/4

### PR DESCRIPTION
**Pull request by magicznyleszek**
## Description

Submission Modal was displaying wrong labels for choice type questions (e.g. `select_one`, `rank`, `score`). It happened whenever multiple choices had the same `name` property.

## Related issues

Fixes #2971
Fixes #2969

### CODE REVIEW

File: jsapp/js/components/submissionDataTable.es6 [hunk 1]
```diff
@@ -124,24 +125,34 @@ class SubmissionDataTable extends React.Component {
...
-        choice = this.findChoice(data);
-        return (
-          <bem.SubmissionDataTable__value>
-            {choice.label[this.props.translationIndex]}
-          </bem.SubmissionDataTable__value>
-        );
+        choice = this.findChoice(listName, data);
+        if (!choice) {
+          console.error(`Choice not found for "${listName}" and "${data}".`);
+          return null;
```
Comment (jnm) [inline jsapp/js/components/submissionDataTable.es6:131]: @magicznyleszek, quick question: if a label for a choice is missing, do we still show the raw label value ("XML name") for the response?

Comment (magicznyleszek) [reply jsapp/js/components/submissionDataTable.es6:131]: Oh, we don't. Good catch!

Comment (magicznyleszek) [reply jsapp/js/components/submissionDataTable.es6:131]: I've added a fallback to `|| choice.name`

**Comment by magicznyleszek** [comment]
@magicznyleszek backport it to `master` pretty please

**Comment by jnm** [comment]
note to self: verify that this fixes a variety of issues described in https://community.kobotoolbox.org/t/data-table-open-view-no-longer-aligned-after-redeployment/15316

**Comment by magicznyleszek** [comment]
Closing this because we want it on `master`: #2980

**Pull request closed without merging**